Platform/ApollolakeBoardPkg/BoardConfig.py

```diff
@@ -95,7 +95,14 @@ def __init__(self, *args, **kwargs):
 			self.STAGE1B_SIZE += 0x2000
 		self.STAGE2_SIZE          = 0x00032000
 		self.PAYLOAD_SIZE         = 0x0001F000
-		self.EPAYLOAD_SIZE        = 0x00120000
+
+		if len(self._PAYLOAD_NAME.split(';')) > 1:
+			# EPAYLOAD is specified
+			self.EPAYLOAD_SIZE      = 0x00120000
```

Comment:
The title says _dynamic_ size, but we are allocating a _maxium_ size. Maybe revise the commit message?

Reply:
It is dynamic according to the payload argument (use epayload or not).